src/StructuredLogViewer/MainWindow.xaml.cs

```diff
@@ -659,31 +663,119 @@ private void Reload()
             OpenLogFile(logFilePath);
         }
 
-        private void SaveAs()
+        private async void RedactSecrets()
         {
-            if (currentBuild != null)
+            RedactInputControl redactInputControl = new RedactInputControl(GetSaveAsDestination);
+            if (redactInputControl.ShowDialog() == true)

```

Comment:
Let's invert this if, and if the result is not true, return early. This would allow to un-indent the rest of the method.